src/connector/src/parser/unified/mod.rs

```diff
@@ -39,7 +39,12 @@ pub type AccessResult<T = Datum> = std::result::Result<T, AccessError>;
 
 /// Access a certain field in an object according to the path
 pub trait Access {
-    fn access(&self, path: &[&str], type_expected: Option<&DataType>) -> AccessResult;
+    fn access(&self, path: &[&str], type_expected: &DataType) -> AccessResult;
+}
+
+/// Whether the whole message is null. This is used for UPSERT events: `null` means DELETE.
```

Comment:
This is not true. For DELETE it is `payload: &[]` rather than
* `&[0x6e, 0x75, 0x6c, 0x6c]` for json, which is `serde_json::Value::Null`
* `&[0x00]` for avro `["null", another_type]`, which is `apache_avro::Value::Union(0, Value::Null)`
* `&[0x02]` for avro `[another_type, "null"]`, which is `apache_avro::Value::Union(1, Value::Null)`

Related: #9826 #12351

Or did we somehow fallback the decoding error of `&[]` (because it is actually not valid json/avro) into a corresponding `Value::Null`? It may also be easy to just test these code paths in a real execution.

Reply:
Well actually I had similar concern. But from the original code (https://github.com/risingwavelabs/risingwave/pull/17032/files#r1620888037), we can say it is DELETE iff `access` returns `None`. Let me double check how it worked before and refine the comments.

Reply:
Yes this is confusing. Basically for DELETE, we should not be able to "access" the payload at all!

Reply:
For JSON: https://github.com/risingwavelabs/risingwave/blob/304b00053c179cb4b0e5f6d9dd0106e01535a8bc/src/connector/src/parser/json_parser.rs#L49-L53

(introduced in https://github.com/risingwavelabs/risingwave/pull/13707/files#diff-ff82fd80555dde971ed6980245ce6154c5e423b06b983cf44a8f11adef372994, how does it work before?)

Reply:
So the previous JSON code path did fallback `&[]` into `&[0x7b, 0x7d]` so that parsing can succeed and output `serde_json::Value::Object(_empty_)`.

Reply:
Btw this may also mean empty json object is wrongly considered as DELETE before